README.md

```diff
@@ -1,100 +1,275 @@
-- There isn't a coverage requirement for this project, beyond demonstrating that every endpoint is covered by some manner of tests.
+| Field            | Datatype | Description
+|------------------|----------|------------
+| `name`           | string   | The customer's name
+| `registered_at`  | string   | `Wed, 29 Apr 2015 07:54:14 -0700`, When the customer first visited the store
+| `address`        | string   | Street address
+| `city`           | string   | &nbsp;
+| `state`          | string   | &nbsp;
+| `postal_code`    | string   | &nbsp;
+| `phone`          | string   | Primary contact phone number
+| `account_credit` | float    | For reason we'd rather not get into, the store owes all of their customers a little bit of money; this amount is made available to customers as credit toward future rentals.
+
+### Testing
+As with all Rails projects, model testing is a requirement. You should have _at least_ one positive and one negative test case for each relation, validation, and custom function you add to your models.
+
+Use good TDD practices, and test _before_ you code. Remember: red-green-refactor.
+
+## Wave 2: Building the API
+In this wave, you will implement the API described below. The endpoints are described more-or-less in order of complexity, and we recommend you build them in that order. Every endpoint must serve JSON data, and must use HTTP response codes to indicate the status of the request.
+
+Because this API will be used as the backend for a future project, there are strict requirements about how it should be structured. To this end, we have provided a simple script that uses HTTParty to exercise all the endpoints. **This script is not a substitute for writing your own tests!!!!!** There are many edge and negative cases it doesn't cover - these are your responsibility.
+
+The schema of your database and the structure of your rails app are completely up to you, so long as the API conforms to the description and provided script.
+
+### Query Parameters
+Any endpoint that returns a list should accept 3 _optional_ query parameters:
+
+| Name   | Value   | Description
+|--------|---------|------------
+| `sort` | string  | Sort objects by this field
+| `n`    | integer | Number of responses to return per page
+| `p`    | integer | Page of responses to return
+
+So, for an API endpoint like `GET /customers`, the following requests should be valid:
+- `GET /customers`: All customers, sorted by ID
+- `GET /customers?sort=name`: All customers, sorted by name
+- `GET /customers?n=10&p=2`: Customers 10-19, sorted by ID
+- `GET /customers?sort=name&n=10&p=2`: Customers 10-19, sorted by name
+
+Things to note:
+- Sorting by ID is the rails default
+- Possible values for `sort` will be specified
+- If the client requests both sorting and pagination, pagination should be relative to the sorted order
+- Check out the [will_paginate gem](https://github.com/mislav/will_paginate)
```

Comment:
OK, maybe sorting and pagination are wave 3 or even optionals. They're not very many lines of code to implement (especially pagination), but it's a lot of testing overhead.